### PR TITLE
Support renaming fields on types deriving `Item`

### PR DESCRIPTION
**Pull request by Phrohdoh**
### What did you implement:

Implemented field renaming on `Item`-derived types via an attribute similar to `serde`'s `rename`.

#### Example

```rust
#[derive(Item, PartialEq, Debug, Clone)]
struct Recipe {
    #[hash]
    #[dynomite(rename = "recipe_id")]
    id: String,
    servings: u64,
}

let value = Recipe {
    id: "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee".into(),
    servings: 2,
};

let attrs: Attributes = value.into();

println!("{:#?}", attrs); // see next code block
```

```
{
    "recipe_id": AttributeValue {
        b: None,
        bool: None,
        bs: None,
        l: None,
        m: None,
        n: None,
        ns: None,
        null: None,
        s: Some(
            "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
        ),
        ss: None,
    },
    "servings": AttributeValue {
        b: None,
        bool: None,
        bs: None,
        l: None,
        m: None,
        n: Some(
            "2",
        ),
        ns: None,
        null: None,
        s: None,
        ss: None,
    },
}
```

Closes: https://github.com/softprops/dynomite/issues/53

### How did you verify your change:

<details>
<summary>Added and ran a new test, field_rename, in derived.rs</summary>

```shell
$ cargo t field_rename
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running target/debug/deps/dynomite-ae35a6724a5f60c6

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 36 filtered out

     Running target/debug/deps/derived-b5a5da7206a8af0c

running 1 test
test tests::field_rename ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 2 filtered out

     Running target/debug/deps/dynomite_derive-050c7f0267e4327d

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

</details>

<details>
<summary>Used `cargo expand` to verify derive output</summary>

```rust
struct Recipe {
    #[hash]
    #[dynomite(rename = "recipe_id")]
    id: String,
    servings: u64,
}
impl ::dynomite::FromAttributes for Recipe {
    fn from_attrs(mut attrs: ::dynomite::Attributes) -> Result<Self, ::dynomite::AttributeError> {
        Ok(Self {
            id: ::dynomite::Attribute::from_attr(attrs.remove("recipe_id").ok_or(
                ::dynomite::AttributeError::MissingField {
                    name: "recipe_id".to_string(),
                },
            )?)?,
            servings: ::dynomite::Attribute::from_attr(attrs.remove("servings").ok_or(
                ::dynomite::AttributeError::MissingField {
                    name: "servings".to_string(),
                },
            )?)?,
        })
    }
}
impl ::std::convert::From<Recipe> for ::dynomite::Attributes {
    fn from(item: Recipe) -> Self {
        let mut values = Self::new();
        values.insert(
            "recipe_id".to_string(),
            ::dynomite::Attribute::into_attr(item.id),
        );
        values.insert(
            "servings".to_string(),
            ::dynomite::Attribute::into_attr(item.servings),
        );
        values
    }
}
impl ::dynomite::Item for Recipe {
    fn key(&self) -> ::std::collections::HashMap<String, ::dynomite::dynamodb::AttributeValue> {
        let mut keys = ::std::collections::HashMap::new();
        keys.insert(
            "recipe_id".to_string(),
            ::dynomite::Attribute::into_attr(self.id.clone()),
        );
        keys
    }
}
struct RecipeKey {
    recipe_id: String,
}
impl ::dynomite::FromAttributes for RecipeKey {
    fn from_attrs(mut attrs: ::dynomite::Attributes) -> Result<Self, ::dynomite::AttributeError> {
        Ok(Self {
            recipe_id: ::dynomite::Attribute::from_attr(attrs.remove("recipe_id").ok_or(
                ::dynomite::AttributeError::MissingField {
                    name: "recipe_id".to_string(),
                },
            )?)?,
        })
    }
}
impl ::std::convert::From<RecipeKey> for ::dynomite::Attributes {
    fn from(item: RecipeKey) -> Self {
        let mut values = Self::new();
        values.insert(
            "recipe_id".to_string(),
            ::dynomite::Attribute::into_attr(item.recipe_id),
        );
        values
    }
}
```

</details>

### What (if anything) would need to be called out in the CHANGELOG for the next release:

Mention the new functionality!

### Potential Follow-Up Work

`#[dynomite(rename_all = "...")]` similar to `serde`'s `rename_all`.

### Other Notes

~~This depends on #66 (and as such includes the commit from it).~~ #66 has been merged and this was rebased on top of it.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
@softprops This is ready for an initial review (and if found acceptable could be made officially reviewable).  I only marked it as a draft PR so the commit messages could be changed, changelog entries could be added, any squashing/rebasing performed, etc.

**Comment by softprops** [comment]
`Phrohdoh` thanks so much for this this looks great!

File: dynomite-derive/src/lib.rs [hunk 1]
```diff
@@ -184,18 +186,74 @@ fn get_to_attribute_map_trait(
     }
 }
 
+/// Get `v` from `#[dynomite(n = v)]`
+fn get_dynomite_name_eq_value_attribute_value_lit(
```
Comment (Phrohdoh) [inline dynomite-derive/src/lib.rs]: I know this is a pretty verbose and ugly name.
Suggestions welcome!

**Comment by Phrohdoh** [comment]
@softprops this is _mostly_ ready for an approvable-review, just need to know how you want the version number changed (I _assume_ bump _minor_, so `0.5.1` -> `0.6`) so I can update that and the changelog.

File: dynomite-derive/src/lib.rs (hunk 1 above)
Comment (softprops) [reply dynomite-derive/src/lib.rs]: I'm much less opinionated about private method names than I am about public ones

**Comment by softprops** [comment]
@Phrohdoh this looks great. I'm happy merge. Let me know if you had any other in-flight changes. I can cut a new release this weekend

**Comment by Phrohdoh** [comment]
@softprops I cherry-picked the commit that changes this to use `syn::Result` instead of `panic!`ing.

Anything else can be done in a follow-up if desired.  :)

Thanks for reviewing!

**Comment by softprops** [comment]
Awesome work. I'll cut a release this weekend

**Pull request merged**